src/index.js

```diff
@@ -30,32 +30,56 @@ class RpcEngine extends SafeEventEmitter {
   // Private
   //
 
-  async _handleBatch (reqs, cb) {
-    const batchRes = []
-    for (const r of reqs) {
-      try {
-        let [err, res] = await this._promiseHandle(r)
-        if (!res) {
-          if (err) {
-            throw err
-          } else {
-            throw ethErrors.rpc.internal('JsonRpcEngine: Request handler returned neither error nor response.')
-          }
-        } else {
-          batchRes.push(res)
-        }
-      } catch (_err) {
-        // some kind of fatal error
-        return cb(_err, null)
+  _handleBatch (reqs, cb) {
+
+    let numRemaining = reqs.length // keep track of number of remaining requests
+    const batchRes = new Array(reqs.length).fill(null) // responses, in order
+
+    // wrap request sending in a promise that resolves when all requests have
+    // been handled
+    new Promise(() => {
```

Comment:
This promise doesn't seem to do anything 🤔 

What exactly is it supposed to do? I suspect removing it would result in the exact same behavior.

Reply:
Hah, there was a `resolve` in there once upon a time, but no longer. Oversight on my part, thanks for catching!